### ISSUE TITLE
Compilation errors with msvc

### ISSUE THREAD

**Issue by eidetic-av** [eyalz800/zpp_bits#72]
Hi, I'm getting a bunch of errors trying to compile on Windows. Does this lib support native Windows builds?

**Comment by eyalz800** [comment]
@eidetic-av I don't remember compiling on msvc but I can offer the following solutions -
1. If you must workaround the issue immediately, you can instruct visual studio to use clang instead 
2. Since I don't have msvc you can send me the errors and I fix them hopefully live.
3. You can give me the code and I try to work this out with compiler explorer

**Comment by eidetic-av** [comment]
Thanks for the quick response. I didn't think of using clang -- I'll give it a go. 
In the meantime, here are the errors I'm seeing:
```
Microsoft (R) Build Engine version 16.11.0+0538acc04 for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

  pointreceiver.cc
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include\chrono(2882,9): warning C4530: C++ exception handler used, but unwind semantics are not enabled. Specify /EHsc [C:\bob\bob-pointreceiver\native\build\pointreceiver.vcxproj]
C:\bob\bob-pointreceiver\native\build\vcpkg_installed\x64-windows\include\spdlog/tweakme.h(74,1): warning C4005: 'SPDLOG_FMT_EXTERNAL': macro redefinition [C:\bob\bob-pointreceiver\native\build\pointreceiver.vcxproj]
C:\bob\bob-pointreceiver\native\src\pointreceiver.cc : message : see previous definition of 'SPDLOG_FMT_EXTERNAL' [C:\bob\bob-pointreceiver\native\build\pointreceiver.vcxproj]
C:\bob\bob-pointreceiver\native\build\vcpkg_installed\x64-windows\include\zpp_bits.h(2438,39): error C2059: syntax error: 'requires' [C:\bob\bob-pointreceiver\native\build\pointreceiver.vcxproj]
C:\bob\bob-pointreceiver\native\build\vcpkg_installed\x64-windows\include\zpp_bits.h(3033): message : see reference to class template instantiation 'zpp::bits::in<std::vector<ByteType,std::allocator<ByteType>>>' being compiled [C:\bob\bob-pointreceiver\native\build\pointreceiver.vcxproj]
          with
          [
              ByteType=std::byte
          ]
C:\bob\bob-pointreceiver\native\src\pointreceiver.cc(20): message : see reference to function template instantiation 'auto zpp::bits::data_in_out<std::byte,>(void)' being compiled [C:\bob\bob-pointreceiver\native\build\pointreceiver.vcxproj]
C:\bob\bob-pointreceiver\native\build\vcpkg_installed\x64-windows\include\zpp_bits.h(2438,37): error C2789: 'zpp::bits::in<std::vector<ByteType,std::allocator<ByteType>>>::resizable': an object of const-qualified type must be initialized [C:\bob\bob-pointreceiver\native\build\pointreceiver.vcxproj]
          with
          [
              ByteType=std::byte
          ]
C:\bob\bob-pointreceiver\native\build\vcpkg_installed\x64-windows\include\zpp_bits.h(2438): message : see declaration of 'zpp::bits::in<std::vector<ByteType,std::allocator<ByteType>>>::resizable' [C:\bob\bob-pointreceiver\native\build\pointreceiver.vcxproj]
          with
          [
              ByteType=std::byte
          ]
C:\bob\bob-pointreceiver\native\build\vcpkg_installed\x64-windows\include\zpp_bits.h(2441,6): error C2143: syntax error: missing ';' before '{' [C:\bob\bob-pointreceiver\native\build\pointreceiver.vcxproj]
C:\bob\bob-pointreceiver\native\build\vcpkg_installed\x64-windows\include\zpp_bits.h(2441,6): error C2334: unexpected token(s) preceding '{'; skipping apparent function body [C:\bob\bob-pointreceiver\native\build\pointreceiver.vcxproj]
C:\bob\bob-pointreceiver\native\build\vcpkg_installed\x64-windows\include\zpp_bits.h(1797,39): error C2059: syntax error: 'requires' [C:\bob\bob-pointreceiver\native\build\pointreceiver.vcxproj]
C:\bob\bob-pointreceiver\native\build\vcpkg_installed\x64-windows\include\zpp_bits.h(2291): message : see reference to class template instantiation 'zpp::bits::basic_out<ByteView>' being compiled [C:\bob\bob-pointreceiver\native\build\pointreceiver.vcxproj]
          with
          [
              ByteView=std::vector<std::byte,std::allocator<std::byte>>
          ]
C:\bob\bob-pointreceiver\native\build\vcpkg_installed\x64-windows\include\zpp_bits.h(3033): message : see reference to class template instantiation 'zpp::bits::out<std::vector<ByteType,std::allocator<ByteType>>>' being compiled [C:\bob\bob-pointreceiver\native\build\pointreceiver.vcxproj]
          with
          [
              ByteType=std::byte
          ]
C:\bob\bob-pointreceiver\native\build\vcpkg_installed\x64-windows\include\zpp_bits.h(1797,37): error C2789: 'zpp::bits::basic_out<ByteView>::resizable': an object of const-qualified type must be initialized [C:\bob\bob-pointreceiver\native\build\pointreceiver.vcxproj]
          with
          [
              ByteView=std::vector<std::byte,std::allocator<std::byte>>
          ]
C:\bob\bob-pointreceiver\native\build\vcpkg_installed\x64-windows\include\zpp_bits.h(1797): message : see declaration of 'zpp::bits::basic_out<ByteView>::resizable' [C:\bob\bob-pointreceiver\native\build\pointreceiver.vcxproj]
          with
          [
              ByteView=std::vector<std::byte,std::allocator<std::byte>>
          ]
C:\bob\bob-pointreceiver\native\build\vcpkg_installed\x64-windows\include\zpp_bits.h(1800,6): error C2143: syntax error: missing ';' before '{' [C:\bob\bob-pointreceiver\native\build\pointreceiver.vcxproj]
C:\bob\bob-pointreceiver\native\build\vcpkg_installed\x64-windows\include\zpp_bits.h(1800,6): error C2334: unexpected token(s) preceding '{'; skipping apparent function body [C:\bob\bob-pointreceiver\native\build\pointreceiver.vcxproj]

```

**Comment by eyalz800** [comment]
I got it, are you compiling with C++20 standard?

**Comment by eidetic-av** [comment]
Oh wow, I feel so dumb haha.

I thought specifying the CXX standard in my CMakeLists would be enough, but 'cmake --build' was defaulting to using an older VS generator.

Thanks for the sanity check... I'll close this issue.

**Comment by eyalz800** [comment]
Ok, let me know if you have any feedback about the library :)

**Comment by eidetic-av** [comment]
Hey @eyalz800, I have a little update. I have since switched to using clang to build for Windows using Visual Studio 2022, and I had an error due to the zpp_bits library. [This line here](https://github.com/eyalz800/zpp_bits/blob/c5e15ec5427c6009c52d0f94fbe62576831fa463/zpp_bits.h#L5375) was causing the error `expected ';' at end of declaration`. Strangely, all I had to do was replace the `xor` with `^` and Visual Studio was happy. I don't know if Visual Studio has an issue with the full word keywords...
